utilities/include/alps/numeric/tensors/tensor_base.hpp

```diff
@@ -108,22 +133,42 @@ namespace alps {
          */
         tensor_base(Container &&container, const std::array < size_t, Dim >& sizes) : storage_(container), shape_(sizes) {
           static_assert(is_storage< T, Container>::value, "Should be either data_storage or data_view type");
+          assert(storage_.size() == size());
```

Comment:
Because you need to know what are you doing. Having additional condition for release mode is slow. 

You have the same in Eigen, it checks  boundaries, correct dimensions and so on  only in debug mode.